libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/tasks/TasksProvider.kt

```diff
@@ -87,14 +88,43 @@ internal open class KotlinTasksProvider {
         compilation: KotlinCompilationData<*>,
         configureAction: (KotlinCompile) -> (Unit)
     ): TaskProvider<out KotlinCompile> {
+        registerTransformsOnceForKotlinCompile(project)
+
         val properties = PropertiesProvider(project)
         val taskClass = taskOrWorkersTask<KotlinCompile, KotlinCompileWithWorkers>(properties)
-        val result = project.registerTask(name, taskClass, constructorArgs = listOf(compilation.kotlinOptions)) {
+        val kotlinCompile = project.registerTask(name, taskClass, constructorArgs = listOf(compilation.kotlinOptions))
+        val classpathConfiguration = if (properties.useClasspathSnapshot) {
+            project.configurations.create("_kgp_internal_${kotlinCompile.name}_classpath").also {
+                project.dependencies.add(it.name, project.files(project.provider { kotlinCompile.get().classpath }))
+            }
+        } else null
+
+        kotlinCompile.configure {
             configureAction(it)
-            KotlinCompile.Configurator(compilation).configure(it)
+            KotlinCompile.Configurator(compilation, properties, classpathConfiguration).configure(it)
+        }
+        configure(kotlinCompile, project, properties, compilation)
+        return kotlinCompile
+    }
+
+    companion object {
+        private const val TRANSFORMS_REGISTERED = "_kgp_internal_kotlin_compile_transforms_registered"
+    }
+
+    private fun registerTransformsOnceForKotlinCompile(project: Project) {
```

Comment:
Can you move configuration&transform creation to a companion object of `KotlinCompile.Configurator`?

Reply:
Done